src/frontend/http.rs

```diff
@@ -596,9 +621,19 @@ pub fn filters(
         .then(upload)
         .map(into_response);
 
+    // Health-check/readiness probe
+    let ctx = context;
+    let health_route = warp::path!("readyz")
+        .and(warp::path::end())
+        .and(warp::get())
+        .and(warp::any().map(move || ctx.clone()))
+        .then(health_endpoint)
+        .map(into_response);
+
     cached_read_query_route
         .or(uncached_read_write_query_route)
         .or(upload_route)
+        .or(health_route)
```

Comment:
Maybe this should be a separate frontend, since the user can have a gRPC frontend enabled without HTTP. But then gRPC isn't authed and HTTP is, so it doesn't make it less secure.

Reply:
Yeah, prefer to leave it there for now, but if there's a strong need can separate it out later on.